### ISSUE TITLE
Restore a model without all x ray lines (EDS)

### ISSUE THREAD

**Issue by ZanettaPM** [hyperspy/hyperspy#1526]
Hello, 
i think an issue occur when i try to create a model with only few x ray lines (not all that are contained in my metadata and not all the family lines), the multifit give good results. So i would like to save and restore my model (big data). But the problem occur here, restore the model is impossible, hyperspy try to load parameters (from dictionary i guess) for all xray lines even the x ray lines removed (from components before the multifit) previously. And for xray lines that have been removed hyperspy doesn't find, of course, any parameters.
I try to save the dictionary after the fit and to load the model with this new one but the same error occur. 
So maybe when i try to restore my model a new one is created with the argument auto_add_lines=True without option to change that. I don't know. 
And thank you for your work !

**Comment by thomasaarholt** [comment]
Could you please give us an example set of code to work with? It doesn't need to be on your data, just so that we can try to understand where things go wrong.

**Comment by ZanettaPM** [comment]
````
# -*- coding: utf-8 -*-
%matplotlib qt
import numpy as np
import tifffile as tiff
import matplotlib.pyplot as plt
import hyperspy.api as hs

from urllib.request import urlretrieve, urlopen
fnames = ['Ni_superalloy_0'+str(i)+'.rpl' for i in range(10,13)] + \
            ['Ni_superalloy_0'+str(i)+'.raw' for i in range(10,13)] + \
            ['Ni_superalloy_0'+str(i)+'.tif' for i in range(10,13)] + \
            ['Ni_superalloy_1pix.msa', 'Ni_La_intensity.hdf5']
url = 'http://cook.msm.cam.ac.uk//~hyperspy//EDS_tutorial//'
files = [urlretrieve(url + fname, fname) for fname in fnames]

s = hs.load("Ni_superalloy_0*.rpl", stack=True)
s.metadata.General.title = '3D EDS map'
s

s = s.as_signal1D(0)
s

s.set_signal_type('EDS_SEM')
s

s1p = hs.load('Ni_superalloy_1pix.msa', signal_type='EDS_SEM')

s.get_calibration_from(s1p)
s.axes_manager[-1].name = 'Energy'

s.metadata

names = ['x', 'y', 'z']
units = '${\mu}m$'
scales = [0.05, 0.05, 0.1]

s.axes_manager.gui()

for i in range(3):
    ax = s.axes_manager[i]
    ax.name = names[i]
    ax.units = units
    ax.scale = scales[i]  

elements = ['Al', 'C', 'Co', 'Cr', 'Mo', 'Ni', 'Ta', 'Ti']
s.set_elements(elements)

s.plot(True)

s.plot() #choose only a part of the signal in order to have a quick multifit
rect_roi=hs.roi.RectangularROI(left=0.15,right=0.25,top=0.1,bottom=1)
roi2D=rect_roi.interactive(s,color="blue")

sr=roi2D

m = sr.create_model()
m.plot()

m.calibrate_energy_axis(calibrate='resolution')
m.calibrate_energy_axis(calibrate='scale')
m.calibrate_energy_axis(calibrate='offset')

m.components
m.remove([2,5,6,7,8,10,11,12,14,15,16,18,20,21,22,23,24,26,28,29,30,31,34,38,39,40,42,43,44]) #randomly removed for the exemple

m.components
m.fit_background(fitter="leastsq",kind='multi',bounded=True)

m.multifit(bounded=True) #fit is really bad but it's also for the exemple

m.save('goodfit.hdf5', 'model') 

u=hs.load('goodfit.hdf5') #load the file works well, we can plot the siganl etc. and the model is well stored

m =u.models.restore('model') #but i can't restore it

````

**Comment by ZanettaPM** [comment]
With Ni data from hyperspy demo, i obtain the same thing. and i write the code with same commands that mine so if it's me the problem we should see it quickly ;)

**Comment by francisco-dlp** [comment]
Thanks for reporting this bug. A fix has just been submitted #1539
